subprojects/execution/src/test/groovy/org/gradle/internal/execution/history/OutputsCleanerTest.groovy

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.execution.history
+
+import org.gradle.api.internal.file.TestFiles
+import org.gradle.internal.file.FileType
+import org.gradle.test.fixtures.file.CleanupTestDirectory
+import org.gradle.test.fixtures.file.TestNameTestDirectoryProvider
+import org.junit.Rule
+import spock.lang.Specification
+
+import java.util.function.Predicate
+
+@CleanupTestDirectory
+class OutputsCleanerTest extends Specification {
```

Comment:
Great that we have some tests for `OutputsCleaner` now! They describe what it does. It would be great to have some Javadoc on the class itself to describe why it does it.